### PR TITLE
Feature/ch11603/deploysmartcontract

### PR DESCRIPTION
**Pull request by richardbahari**
This work is related to ticket 11603, to deploy smart contract with basic values.

Currently there is an example created on SDK named SDKCreateSmartContract. This would run a deployment to the ETH.

I have had to create some factory classes to deal with the ABI datatypes from web3j, as they are needed to instantiate those objects when calling FunctionEncoder.encodeConstructor. This is quite different to the way FunctionEncoder.makeFunction works. The latter seems happy to receive the types and behind the scene it will try to instantiate.

But, the encodeConstructor requires the actual ABI datatype instantiated in the call, so I can't just call it encodeConstructor({uint, byte32}, ... , {1, 10}...) like makeFunction.

to clarify, see example on how the call is expected:
String encodedConstructor = FunctionEncoder.encodeConstructor(Arrays.asList(new Bool(true),
                                new Int256(5),
                                new Uint16(33),
                                new Byte32("0x68656c6c6f"),
                                new Address("0x650A87cfB9165C9F4Ccc7B971D971f50f753e761"),
                                new Utf8String("Hi_there!"),
                                boolArray));

### CODE REVIEW

File: overledger-sdk-essential/src/main/java/network/quant/essential/dto/DltTransactionRequest.java [hunk 1]
```diff
@@ -30,4 +30,5 @@
     @JsonIgnore
     Long sequence;
 
+
 }
```
Comment (kashif-quant) [inline overledger-sdk-essential/src/main/java/network/quant/essential/dto/DltTransactionRequest.java:34]: No change here, can we remove this file from the change list?

File: overledger-sdk-essential/src/test/java/network/quant/essential/AccountManagerTest.java [hunk 2]
```diff
@@ -44,7 +44,16 @@ public void sign(String fromAddress, String toAddress, String message, DltTransa
...
+            public void invokeContract(DltTransaction dltTransaction){}
+
+            @Override
+            public void createSmartContract(DltTransaction dltTransaction) {
 
+            }
+
+            @Override
+            public DltTransaction buildSmartContractQuery(DltTransaction dltTransaction){return null;}
             @Override
```
Comment (kashif-quant) [inline overledger-sdk-essential/src/test/java/network/quant/essential/AccountManagerTest.java:57]: Do we need a removeSmartContract() as well?

Comment (richardbahari) [reply overledger-sdk-essential/src/test/java/network/quant/essential/AccountManagerTest.java:57]: probably not. I read somewhere that the contracts could be removed by supporting a self destruct call when created. Could be a question to Luke and Tash.

**Comment by najla-quant** [comment]
@richardbahari @mahsa-bytt @kashif-quant  I think your branch has been branched off Mahsa's one feature/ch11529/invokesmartcontract if I am not wrong ? 
If so, it should be better to approve and merge Masha's branch and then review your branch (we will have less code).
Mahsa is now checking the security packages to be ran with @SalmanQuantNetwork

**Comment by richardbahari** [comment]
@najla-quant  yes thats right, yes its better to have Mahsa's code to be merged first. Once that is merged to develop, i could rebase develop against my feature branch here

> @richardbahari @mahsa-bytt @kashif-quant I think your branch has been branched off Mahsa's one feature/ch11529/invokesmartcontract if I am not wrong ?
> If so, it should be better to approve and merge Masha's branch and then review your branch (we will have less code).
> Mahsa is now checking the security packages to be ran with @SalmanQuantNetwork

**Comment by richardbahari** [comment]
I am closing this PR and will raise a new one as Mahsa's PR was merged already, I will rebase and raise a new one soon.

**Pull request closed without merging**